### PR TITLE
[saml2-js] Fix type of SAMLAssertResponse['response_header']['id']

### PR DESCRIPTION
**Pull request by dbeckwith**
The type of the `SAMLAssertResponse['response_header']['id']` field was mistakenly set to `'string'` (a type that only accepts the string literal "string") instead of `string` (a type that accepts any string).

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## saml2-js ([<kbd>unpkg</kbd>](https://unpkg.com/browse/saml2-js@latest/))
was missing the following properties:
1. create_authn_request
1. sign_authn_request
1. create_metadata
1. format_pem
1. sign_request

<details>
<summary>as well as these 12 other properties...</summary>
<p>check_saml_signature, check_status_success, pretty_assertion_attributes, decrypt_assertion, parse_response_header, parse_logout_request, get_name_id, get_session_info, parse_assertion_attributes, add_namespaces_to_child_assertions, set_option_defaults, extract_certificate_data</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 543bc3df0da0e4eb7db919a5b3e6e4accc5c3fc3
</p>

**Review by jgeurts — APPROVED** [review]

**Comment by jgeurts** [comment]
LGTM!

**Comment by dbeckwith** [comment]
Had to update the tests to satisfy the bot (it said a maintainer has to approve if I don't modify tests)

**Comment by dbeckwith** [comment]
@jgeurts Could you take another quick look at this when you get a chance? I just had to add some tests.

**Review by jgeurts — APPROVED** [review]

**Pull request merged**
